custom_components/candy/client/model.py

```diff
@@ -181,28 +185,279 @@ class OvenState(StatusCode):
+
+
+@dataclass
+class HoodStatus:
+    machine_state: HoodState
+    light: bool
+    fan: bool
+    grease_filter: bool
+    carbon_Filter: bool
+    warning: bool
+    remote_control: bool
+
+    @classmethod
+    def from_json(cls, json):
+        return cls(
+            machine_state=HoodStatus.parse_state(json),
+            light=json["Light"] == "1",
+            fan=int(json["Fan"]) > 0,
+            grease_filter=json["GreaseFilter"] == "1",
+            carbon_Filter=json["CarbonFilter"] == "1",
+            warning=json["Warning"] == "1",
+            remote_control=json["WiFiStatus"] == "1",
+        )
+
+    @staticmethod
+    def parse_state(json) -> str:
+        """
+        Parse Hood status
+        """
+        if int(json["Fan"]) > 0 and json["Light"] == "1":
+            return HoodState.FAN_LIGHT
+
+        if int(json["Fan"]) > 0:
+            return HoodState.FAN
+
+        if json["Light"] == "1":
+            return HoodState.LIGHT
+        else:
+            return HoodState.IDLE
+
+
+class FridgeState(StatusCode):
+    NORMAL = (0, "Normal")
+    ECO = (1, "Eco")
+    SUPER_FREEZING = (2, "Super freezing")
+    SMART_COOLING = (3, "Smart cooling")
+    SUPER_FREEZING_COOLING = (4, "Super freezing & cooling")
+    ERROR = (5, "Error")
+
+
+@dataclass
+class FridgeStatus:
+    machine_state: HobState
+    cooling_temperature: bool
+    freezing_temperature: bool
+    eco_mode: bool
+    super_freezing_mode: bool
+    smart_cooling_mode: bool
+    door_locked: bool
+    door_open: bool
+    fan: int
+    error: Optional[int]
+    remote_control: bool
+
+    @classmethod
+    def from_json(cls, json):
+        return cls(
+            machine_state=FridgeStatus.parse_state(json),
+            cooling_temperature=int(json["FrSet"]),
+            freezing_temperature=int(json["FzSet"]),
+            eco_mode=json["Eco"] == "1",
+            super_freezing_mode=json["Spr"] == "1",
+            smart_cooling_mode=json["Ice"] == "1",
+            door_locked=json["Lck"] == "1",
+            door_open=json["Door"] == "1",
+            fan=int(json["Fan"]),
+            error=int(json["Error"]) if int(json["Error"]) != 0 in json else None,
+            remote_control=json["WiFi"] == "1",
+        )
+
+    @staticmethod
+    def parse_state(json) -> str:
+        """
+        Parse Fridge status
+        """
+
+        if json["Error"] != "0":
+            return FridgeState.ERROR
+
+        if json["Eco"] == "1":
+            return FridgeState.ECO
+
+        if json["Ice"] == "1" and json["Spr"] == "1":
+            return FridgeState.SUPER_FREEZING_COOLING
+
+        if json["Ice"] == "1":
+            return FridgeState.SUPER_FREEZING
+
+        if json["Spr"] == "1":
+            return FridgeState.SMART_COOLING
+        else:
+            return FridgeState.NORMAL
+
+
+def only_numerics(seq):
+    seq_type = type(seq)
+    return seq_type().join(filter(seq_type.isdigit, seq))
+
+
+def oven_remaining_minutes(time):
```

Comment:
Same here, an explanation of why this is needed would help future me :)

Reply:
you men oven_remaining_minutes?
Oven have remaining time. But this Candy Api is strange in some cases. In recive remaining minutes for oven of value 1092 but at the same time oven was not heating. I decided to add this exception whic show remaining time correct (actually it is very trange that user turn on oven for more than 1092 minutes....